stdlib/src/memory/maybe_uninitialized.mojo

```diff
@@ -0,0 +1,77 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+
+# TODO: Make this public when we are certain of the design.
+# TODO: Move _size into an alias when the bug https://github.com/modularml/mojo/issues/2889
+# is fixed.
+struct _MaybeUninitialized[ElementType: CollectionElement, _size: Int = 1](
```

Comment:
```suggestion
struct MaybeUninitialized[ElementType: CollectionElement](
```

I think we can call this `MaybeUninitialized`. I think we we will want something very similar to this, and we don't have backwards compatibility to worry about yet in the stdlib. So lets avoid the churn🙂

Reply:
Great to know, I'll change it.